ci/kind/kind-setup.sh

```diff
@@ -517,18 +517,20 @@ function destroy_external_servers {
 
 function clean_kind {
     echo "=== Cleaning up stale kind clusters ==="
-    read -a all_kind_clusters <<< $(kind get clusters)
-    for kind_cluster_name in "${all_kind_clusters[@]}"; do
-        creationTimestamp=$(kubectl get nodes --context kind-$kind_cluster_name -o json -l node-role.kubernetes.io/control-plane | \
+    for context in $(kubectl config get-contexts -o name | grep 'kind-'); do
+      cluster_name=$(echo $context | sed 's/^kind-//')
+      if docker ps --format '{{.Names}}' | grep -q "$cluster_name"; then
```

Comment:
Does this list all containers in the given cluster name ,  `docker ps --format '{{.Names}}' | grep -q "$cluster_name"` ?
`docker ps --format '{{.Names}}' | grep 'multi' `
```suggestion
      if docker ps --format '{{.Names}}' | grep 'multi'; then
```

Reply:
We don't need to list the cluster names we just want to check if they are present or not, so -q is required here.